### DISCUSSION TITLE
How to cleanly exit after `-h, --help`?

### DISCUSSION THREAD

**Discussion by kkerce** [ko1nksm/getoptions#56] [Q&A] [answered]
In [this example](https://github.com/ko1nksm/getoptions/blob/a6c2df16302a2d0cd4cafb5396ef3421b2c3a1aa/examples/basic.sh#L27), can anyone please provide an example of how to exit (with no error status) if `-h` or `--help` was supplied to the program, preferably without a hack involving a global variable?  I must be overlooking something obvious.

**Comment by ko1nksm** [comment, marked as answer]
It should exit with no error status.

![image](https://github.com/user-attachments/assets/9e052752-24e5-47c5-8272-f6081c4743f8)

**Comment by kkerce** [comment]
My apologies @ko1nksm, I was using the changes proposed in #53.  As it turns out, [one of the commits from that PR](https://github.com/ko1nksm/getoptions/pull/53/commits/99f9b411ac8921ee9e12c2d09919724f2000d4d8) breaks the `-h, --help` aspect we discussed above.  I'll point out the bug in a comment in #53.  I would love to be able to use getoptions in functions.

**Reply by ko1nksm** [reply]
Ok. I expected #53 would break something, but was not yet aware of the problem. It's help.
